clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -1085,6 +1085,68 @@ def CmpOp : CIR_Op<"cmp", [Pure, SameTypeOperands]> {
   let hasVerifier = 0;
 }
 
+//===----------------------------------------------------------------------===//
+// CheckedArithOp
+//===----------------------------------------------------------------------===//
+
+def CAK_Add : I32EnumAttrCase<"Add", 1, "add">;
+def CAK_Sub : I32EnumAttrCase<"Sub", 2, "sub">;
+def CAK_Mul : I32EnumAttrCase<"Mul", 3, "mul">;
+
+def CheckedArithKind : I32EnumAttr<
+    "CheckedArithKind",
+    "checked arithmetic operation kind",
+    [CAK_Add, CAK_Sub, CAK_Mul]> {
+  let cppNamespace = "::mlir::cir";
+}
+
+def CheckedArithOp : CIR_Op<"checked_arith", [Pure, SameTypeOperands]> {
```

Comment:
Naming: might be nice to be a bit closer to binop. I suggest `cir.binop.checked` or `cir.binop.overflow`. I think I like `overflow` better, since it's more obvious of what it is checking, but I'm open.

Reply:
Updated.